include/aos/common/tools/string.hpp

```diff
@@ -187,7 +180,7 @@ class String : public Array<char> {
      * @param srt string to compare with.
      * @return bool.
      */
-    friend bool operator==(const char* cStr, const String& str) { return str.operator==(cStr); };
+    friend bool operator==(const char* cStr, const String& str) { return str == String(cStr); };
```

Comment:
It looks like we only need operators ==/!= (const String&, const String&) and const char* would be implicitly converted to to String to match the operand type.